pageserver/benches/bench_layer_map.rs

```diff
@@ -203,5664 +213,46 @@ fn large_layer_map(c: &mut Criterion) {
             result.unwrap();
         });
     });
+
+    // TODO I don't think I'm using criterion right. Should I keep adding to the
+    //      same function?
+
+    // A list of layer filenames, extracted from a real project that was taknig
+    // too long processing layer map queries.
+    //
+    // TODO consider compressing these files
+    let layer_map = build_layer_map(PathBuf::from("benches/odd-brook-layernames.txt"));
+
+    // We'll query the midpoint of each image layer at LSN right before the image
+    // layer. This way we get a somewhat uniform coverage of both the lsn and key
+    // space.
+    let queries: Vec<(Key, Lsn)> = layer_map
+        .iter_historic_layers()
+        .filter_map(|l| {
+            if l.is_incremental() {
+                None
+            } else {
+                let kr = l.get_key_range();
+                let lr = l.get_lsn_range();
+
+                let mid_key = kr.start; // TODO maybe actually find the midpoint
```

Comment:
Just `kr.start/kr.end` will not work and find median point seems too be too expensive...
But right now name `mid_key` is confusing

Reply:
Yes, I plan to fix this TODO before merging